Documentation for coverage support

gergelyfabian opened the pull request:
### Description
<!-- Mandatory: A crisp one or two line description of your proposed change. -->
Documentation for coverage support

<!-- Optional:
  A longer explanation of your proposed changes..
  This includes listing any breaking changes, if there are any.
-->

### Motivation
<!-- Mandatory: A summary of why you are making this change. -->
I was specifically asked to add documentation for coverage support :)

d-haxton commented:
>Should we say it’s currently only tested with scalatest?

Based off of my last coverage PR, this should work with junit without any issue as well. Although I goofed and missed an update to the `test-b2` test. I'll put up a PR for that shortly (we can also turn on testing for that which would be good)

> Having users of this ruleset depend on a toolchain from the test sources sounds wrong. Maybe we should move the toolchain to the production code part?

My current approach is to just define a scala_toolchain & toolchain within a build file and just set `enable_code_coverage_aspect = "on"`, although at this point it would seem reasonable to just remove the requirement for toolchain altogether as coverage is just broken without it.

gergelyfabian commented on docs/coverage.md:
````diff
@@ -0,0 +1,54 @@
+bazel coverage --extra_toolchains="@io_bazel_rules_scala//test/coverage:enable_code_coverage_aspect" //...
+```
+
+It will produce several .dat files with results for your targets.
````
Done.

gergelyfabian commented on docs/coverage.md:
````diff
@@ -0,0 +1,54 @@
+lcov --list your-coverage-result.dat
+```
+
+If you prefer an HTML report, then you can use `genhtml` provided also by the `lcov` package.
````
I cannot comment on that in the README as I myself haven't managed to set up IntelliJ with code coverage just yet. I'd be happy to see a later PR to the README with updates on that and try it myself :)

gergelyfabian commented:
> > Should we say it’s currently only tested with scalatest?
> 
> Based off of my last coverage PR, this should work with junit without any issue as well. Although I goofed and missed an update to the `test-b2` test. I'll put up a PR for that shortly (we can also turn on testing for that which would be good)

I have no information on that part. Is the PR merge-able as it is?

> > Having users of this ruleset depend on a toolchain from the test sources sounds wrong. Maybe we should move the toolchain to the production code part?
> 
> My current approach is to just define a scala_toolchain & toolchain within a build file and just set `enable_code_coverage_aspect = "on"`, although at this point it would seem reasonable to just remove the requirement for toolchain altogether as coverage is just broken without it.

I'd support moving the toolchain to the production code part, should I change the PR anyhow for that?

gergelyfabian commented:
1. I changed the description to mention that coverage has only been tested with ScalaTest
2. Added information about combined coverage reports (and changed the example script to reflect that)
3. IntelliJ support: I can not provide any information on that as I haven't worked that out yet
4. Move coverage toolchain to production code: could this change be done in a later separate PR and merge this one to reflect the current state?

gergelyfabian commented:
> Thanks!
> Sorry for the long delay but I was swamped.
> If you'd like to follow up with a PR for the toolchain move to production code I'd appreciate it.

Thank you! Unfortunately I think I don't know how that toolchain move would need to be done (haven't grasped that part of Bazel yet).

ittaiz commented:
No worries, I’ll do it

On Sat, 7 Mar 2020 at 13:06 Gergely Fábián <notifications@github.com> wrote:

> Thanks!
> Sorry for the long delay but I was swamped.
> If you'd like to follow up with a PR for the toolchain move to production
> code I'd appreciate it.
>
> Thank you! Unfortunately I think I don't know how that toolchain move
> would need to be done (haven't grasped that part of Bazel yet).
>
> —
> You are receiving this because you modified the open/close state.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/bazelbuild/rules_scala/pull/1017?email_source=notifications&email_token=AAKQQF5OEDSF4YP2UHHUQY3RGITD3A5CNFSM4K6CG4L2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEODWQMY#issuecomment-596076595>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAKQQF76BRWTY74QPTWK2ATRGITD3ANCNFSM4K6CG4LQ>
> .
>
-- 


*Ittai Zeidman*



40 Hanamal street, Tel Aviv, Israel



<http://www.wix.com>

ittaiz commented:
#1020
